Sitemap.php

```diff
@@ -479,7 +485,7 @@ public function setUseGzip($value)
         if ($value && !extension_loaded('zlib')) {
             throw new \RuntimeException('Zlib extension must be enabled to gzip the sitemap.');
         }
-        if ($this->urlsCount && $value != $this->useGzip) {
+        if (isset($this->writerBackend) && $value != $this->useGzip) {
```

Comment:
`$this->writerBackend !== null` could be used.

Reply:
Ok, seems that would be more in line with the general style.